### ISSUE TITLE
[FEAT REQ] consider-using-defaultdict: Consider use collections.defaultdict instead of setdefault

### ISSUE THREAD

**Issue by moylop260** [PyCQA/pylint#1400]
### Steps to reproduce
1. Use the following code:
```python
d = {}
for name in names:
    key = len(name)
    d.setdefault(key, []).append(name)

```
2. Run pylint

### Current behavior
A message about use `setdefault` is not emitted

### Expected behavior
IMHO It could be a good emit the message:
`consider-using-defaultdict: Consider use collections.defaultdict instead of setdefault`

What do you think about?

**Comment by rogalski** [comment]
I disagree, they are both equally valid styles and I think we should avoid very _nitpicky_ checkers.

**Comment by moylop260** [comment]
I'm not sure if is really a nitpicky checker and a matter of style...

But maybe I'm wrong.

Could you review the Raymond Hettinger's video minute 26:

[![IMAGE ALT TEXT](http://img.youtube.com/vi/OSGv2VnC0go/0.jpg)](https://youtu.be/OSGv2VnC0go?t=26m40s "Video Title")

Executing a small timeit test:

Using `setdefault`:

 - `python -m timeit --setup="d = {};names=map(str, range(100000))" "[d.setdefault(len(name), []).append(name) for name in names]"`
The result was: 10 loops, best of 3: 27.9 msec per loop

Using `defaultdict`:

 - `python -m timeit --setup="from collections import defaultdict;names=map(str, range(100000));d = defaultdict(list)" "[d[len(name)].append(name) for name in names]"`
The result was: 100 loops, best of 3: 15.6 msec per loop

**Comment by PCManticore** [comment]
I agree with @rogalski While I would see the value in this one, I think it is going to be too nitpicky.
